shared/elements/gaia_lowstorage_dialog/script.js

```diff
@@ -0,0 +1,98 @@
+'use strict';
+/* global ComponentUtils, MozActivity */
+
+/**
+ * The gaia-confirm component displays a dialog in which the user has a
+ * choice to confirm or cancel the action. It may be displayed along with a
+ * title, description, and image. Buttons may also be configured.
+ * @requires GaiaButtons
+ */
+window.GaiaLowstorageDialog = (function(win) {
+  // Extend from the HTMLElement prototype
+  var proto = Object.create(HTMLElement.prototype);
+
+  // Allow baseurl to be overridden (used for demo page)
+  var baseurl = window.GaiaLowstorageDialogBaseurl ||
+    '/shared/elements/gaia_lowstorage_dialog/';
+
+  function learnmore() {
+    var activity = new MozActivity({
+      name: 'configure',
+      data: {
+        section: 'applicationStorage'
+      }
+    });
+
+    return Promise.resolve(activity);
+  }
+
+  /**
+   * Localize the component manually as l10n attributes are not supported
+   * within the shadow dom. See also: bug 1026236.
+   */
+  proto._localizeShadowDom = function() {
+    navigator.mozL10n.translateFragment(this.shadowRoot);
+  };
+
+  proto.createdCallback = function() {
+    var shadow = this.createShadowRoot();
+
+    var node = document.importNode(template.content, true);
+    shadow.appendChild(node);
+    ComponentUtils.style.call(this, baseurl);
+
+    navigator.mozL10n.ready(this._localizeShadowDom.bind(this));
```

Comment:
> createdCallback is always called only once but now I realize from your comment that we don't free it when the element is removed. So I think I should attach on "attachedCallback" and detach on detachedCallback (we don't know when the object is garbage collected). But... we can't detach l10n's ready event. So I think I'll either have to use a global event handler and do my own attach/detach mechanism, or implement this in l10n.js.

As discussed with @julienw on IRC, perhaps a safer solution memory-wise would be to use `mozL10n.once()` and `window.addEventListener('localized', …)` inside of it.

Or, we could introduce the l20n-style proxy in l10n.js for `shared/` code, as suggested by @julienw in https://bugzilla.mozilla.org/show_bug.cgi?id=1203108#c7.  Both @zbraniecki  and I like this approach;  we might  be able to land it soon.